engines/bastion_katello/app/assets/javascripts/bastion_katello/content-credentials/content-credential.factory.js

```diff
@@ -7,77 +7,82 @@
  * @description
  *   Provides a BastionResource for Content Credentials.
  */
-angular.module('Bastion.content-credentials').factory('ContentCredential',
-    ['BastionResource', 'CurrentOrganization',
-    function (BastionResource, CurrentOrganization) {
+angular.module("Bastion.content-credentials").factory("ContentCredential", [
+    "BastionResource",
+    "CurrentOrganization",
+    function(BastionResource, CurrentOrganization) {
+        function appendData(collection, fullData, usedAs) {
+            angular.forEach(fullData, function(data) {
+                data.usedAs = usedAs;
+                collection.push(data);
+            });
+        }
 
-        return BastionResource('katello/api/v2/content_credentials/:id/:action',
-            {id: '@id', 'organization_id': CurrentOrganization},
-            {
-                autocomplete: {method: 'GET', isArray: true, params: {id: 'auto_complete_search'}},
-                update: {method: 'PUT'},
-                products: {method: 'GET', transformResponse:
-                    function (data) {
-                        var contentCredential = angular.fromJson(data);
-                        var allProducts = {};
-                        allProducts.length = 0;
-                        allProducts.products = {};
-
-                        function collectProducts(allP, thisP, usedAs) {
-                            var key;
-                            for (key in thisP) {
-                                if (thisP.hasOwnProperty(key)) {
-                                    thisP[key]["used_as"] = usedAs;
-                                    allP.products[allP.length] = thisP[key];
-                                    allP.length++;
-                                }
-                            }
-                        }
-
-                        collectProducts(allProducts, contentCredential.gpg_key_products, "GPG Key");
-                        collectProducts(allProducts, contentCredential.ssl_ca_products, "SSL CA Cert");
-                        collectProducts(allProducts, contentCredential.ssl_client_products, "SSL Client Cert");
-                        collectProducts(allProducts, contentCredential.ssl_key_products, "SSL Client Key");
+        function sortContentCredentialData(data, cases) {
+            var contentCredential = angular.fromJson(data);
+            var allData = [];
+            _.forOwn(contentCredential, function(value, key) {
+                switch (key) {
```

Comment:
I love all of the refactors in here. I have an idea for one more: (afaict) this switch can be eliminated by keeping a map of the `key` ('used as') from internal name to friendly UI name. It also stops us from caring about a specific ordering in `cases` like the switch does

```
var someMap = {
  "gpg_key_products": "GPG Key",
  <snip>
}

appendData(allData, value, someMap[key]);

```

I think it might bring whether we need a separate 'appendData' method into question but I don't feel strongly one way or the other - up to you!